javascript/javascript1/week4/voiceAssistant.js

```diff
@@ -0,0 +1,218 @@
+'use strict'
+const todoList = []
+const userName = []
+const userAge = []
+var splitedCommand = []
+
+// Main function
+function getReply(command) {
+    // we split the command in an array so we can work with it
+    splitedCommand = command.split(' ')
+    //we loop througth the array command
+    for (let i=0; i<splitedCommand.length; i++){   
+       if (splitedCommand[i] === "name"){  
+            return nameFunc(splitedCommand)          
+        }
+       else if(splitedCommand[i] === "todo"){
+           return todoFunc(splitedCommand)
+       }
+       else if (splitedCommand[i] === "day"){
+           return getTodaysDay()
+       }
+       else if (splitedCommand[i] === "+" | splitedCommand[i] === "-" | splitedCommand[i] === "*" | splitedCommand[i] === "/"){
+           return doMath()
+       }
+       else if(splitedCommand[i] === "Set"){
+           return setTimer()
+       }
+       else if(splitedCommand[i] === "age"){
+           return ageFunc()
+       }
+    }
+    return "I don't undestand this command."
+}
+
+// function that handles name greetings and saving
+// We know that it contains the word 'name'
+function nameFunc(splitedCommand){
+    for (let i=0; i<splitedCommand.length; i++){
+        if (splitedCommand[i] === "Hello" | splitedCommand[i] === "hello"){
+            // save user name in variable 'givenName'
+            const givenName = splitedCommand[splitedCommand.length -1]
+            // check if user name saved in array 'userName'
+            if (userName.includes(givenName)){
+                return "hello again "+ givenName
+            }
+            // if its not saved, add it to array 'userName' and replay with nice to meet you + username
+            else{
+                userName.push(givenName)
+                return "nice to meet you "+ givenName
```

Comment:
An alternative method to do this would be as example below

For example, you can split command string into an array of words (relying on spaces)

const commandWords = command.split(' ');
There are many ways to search if the command string includes some words like

  if (command.startsWith('hello my name is')) {
    name = commandWords[commandWords.length - 1];
    return `Nice to meet you ${name}`;
  }
or

  if (command.includes('hello my name is')) {
To get the name you can also replace all characters except the name into an empty string

    const name = command.replace('hello my name is', '');

Reply:
Yes, Thank you for your feedback